src/utils/dump-context.js

```diff
@@ -12,9 +12,24 @@
 
 const path = require('path');
 const fs = require('fs-extra');
+const { setdefault } = require('@adobe/helix-shared').types;
 
-function tstamp() {
-  const now = new Date();
+
+/**
+ * Returns {@code true} if context dumps should never be written to disk.
+ * @returns {boolean} {@code true} if writing dumps is disabled.
+ */
+function disableDisk() {
```

Comment:
I'd use the `production` check we already have, which guards against exceptions when accessing the environment vars directly.

Reply:
- I think `production` is the wrong test for this purpose. I just don't want to dump in openwhisk - no matter if it's production or stage or dev...etc....
- `!!process.env.*` will never thrown an exception (because `process.env` is guaranteed to exist in nodejs)

Reply:
> `!!process.env.*` will never thrown an exception (because `process.env` is guaranteed to exist in nodejs)

you would assume so, yes. but:
https://github.com/adobe/helix-pipeline/issues/142
https://github.com/adobe/helix-cli/issues/408#issuecomment-448621464

Reply:
hmmm.... I think this was a temporary problem....

see my test:
```js
function main(params) {
  return {
    statusCode: 200,
    headers: { 'Content-Type': 'text/html', 'X-Foo': 'bar' },
    body: JSON.stringify(process.env, null, 2),
  };
}
```


```
$ curl https://adobeioruntime.net/api/v1/web/helix-pages/default/test
{
  "__OW_ACTION_NAME": "/helix-pages/test",
  "__OW_ACTIVATION_ID": "9f4a78a9ed2a40818a78a9ed2a60810d",
  "__OW_DEADLINE": "1560327247499",
  "__OW_NAMESPACE": "helix-pages",
  "NODE_VERSION": "10.15.1",
  "HOSTNAME": "a6d134123987",
  "YARN_VERSION": "1.13.0",
  "SHLVL": "1",
  "HOME": "/home/runtimeuser",
  "PATH": "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
  "NOT_ROOT_USER": "runtimeuser",
  "MESOS_SANDBOX": "/mnt/mesos/sandbox",
  "__OW_ALLOW_CONCURRENT": "true",
  "UID": "1001",
  "__OW_API_HOST": "https://runtime.adobe.io:443",
  "__OW_JSON_LOG": "true",
  "PWD": "/nodejsAction",
}
```

Reply:
I looked closer at the stack trace. The error was thrown by code in the openwhisk node runtime:

```javascript
TypeError: Cannot read property '__OW_ACTIVATION_ID' of null
    at Object.get [as __OW_ACTIVATION_ID] (/nodejsAction/src/service.js:251:60)
    at production (/nodejsAction/XXDPCCdJ/node_modules/@adobe/helix-pipeline/src/utils/is-production.js:18:50)
...
``` 
The node runtime injects dynamic properties into `process.env` in order to enable concurrent activations. There must have been a (temporary?) issue.

Reply:
they probably have a special getter for the `process.env`, in order to isolate it for concurrent invocations. maybe it was some (scary) raise-condition... 😨